index.bs

```diff
@@ -16,13 +16,448 @@ WPT Display: closed
+
+    Issue: It is not yet possible to specify how the [=user agent=] is expected to read a [=request=]'s [=request/eligible for storage-access=] boolean when attaching cookies to an outgoing [=request=], since the cookie store has no defined interface that specs can interact with.
+
+    Note: A [=request=] also has a [=request/client=], which is an [=environment settings object=] (a kind of [=environment=]) and therefore has a [=environment/has storage access=] boolean. That value is distinct from the value of the [=request=]'s [=request/eligible for storage-access=] boolean. Both represent an "opt in" signal for accessing unpartitioned cookies in a cross-site context, but the signal comes from different [=origin=]s. The [=request=]'s [=request/client=]'s [=environment/has storage access=] field represents whether the [=site=] [=obtain a site|obtained=] from the [=environment=]'s [=environment settings object/origin=] has opted in. The [=request=]'s [=request/eligible for storage-access=] field represents whether the [=site=] [=obtain a site|obtained=] from the [=request=]'s [=request/url=]'s [=url/origin=] has opted in.
+
+A [=request=] has an associated <dfn for="request">single-hop cache mode</dfn>, whose value is null or a [=request/cache mode=]. 
+
+This document renames a [=request=]'s [=request/cache mode=] field to <dfn for="request">internal cache mode</dfn>.
+
+<div algorithm="request cache mode">
+This document redefines a [=request=]'s <dfn for="request" export>cache mode</dfn> as the cache mode returned by running the following steps, given a [=request=] |request|:
+    1. If |request|'s [=request/single-hop cache mode=] is not null, return |request|'s [=request/single-hop cache mode=].
+    1. Return |request|'s [=request/internal cache mode=].
+</div>
+
+A <dfn>storage access status</dfn> is one of "<dfn for="storage access status">none</dfn>", "<dfn for="storage access status">inactive</dfn>", or "<dfn for="storage access status">active</dfn>".
+
+<div algorithm>
+    The <dfn for=request>storage access status</dfn> of a [=request=] |request| is the [=storage access status=], or null, returned by running the following steps:
+        1. If the user agent's cookie store would attach cookies with the `SameSite=Strict` attribute to |request|, return null. [[!COOKIES]]
+        1. Let |allowed| be a [=boolean=], initially set to the result of determining whether the user agent's cookie store allows unpartitioned cookies to be attached to |request|.
```

Comment:
"determining whether the user agent's cookie store allows unpartitioned cookies" should be a link, right?

Reply:
I would make it so, if it were possible. Fetch's interaction with cookies is largely unspecified (see https://github.com/httpwg/http-extensions/issues/2084), so it's not possible to specify this concretely (as the below issue points out).

Reply:
This document contains the following:

spec: storage-access; urlPrefix: https://privacycg.github.io/storage-access/
     type: dfn
     text: determine whether the user agent explicitly allows unpartitioned cookie access; url: #determine-whether-the-user-agent-explicitly-allows-unpartitioned-cookie-access

And it looks like https://privacycg.github.io/storage-access/ defines a method using that exact string, though I guess it takes two sites, rather than a fetch.  So I guess the method is overloaded, with one overload not really existing?  If there's no actual method, what use is it to set a temporary field before calling an unspecified method?

Reply:
I was imagining that some browser spec (Fetch or HTML) would define a wrapper around the underlying cookie store, and that wrapper would be where algorithms like this live. So I probably wasn't clear; I didn't mean to teach the cookie store itself how to do this algorithm (since that violates the layering), but rather to teach the browser's wrapper about this instead. Does that match what you're thinking?

Even if we're on the same page about that, there's still the question of what the "right" thing to do today is. I can write this as a standalone algorithm that'll eventually have to move and be consolidated with other browser cookie store logic, if you think that's the right approach.

Reply:
Hmm I don't think we've been really designing this with a browser wrapper in mind.

Anyway, writing an algorithm that takes into account all the different states of a request and its originating document to determine third-party cookie blocking sounds out of scope for this work and like something that @dcthetall will soon pick up as part of getting cookie layering into Fetch/HTML, so maybe this specific algorithm is what you can punt on?

Reply:
Based on all the discussion here and in the other thread, I'll write out this algorithm "properly" and reference the other one defined in SAA.